### PR TITLE
Enable multi-chip op testing and add Gemma TP op tests

### PR DESCRIPTION
**Pull request by gengelageTT**
### Ticket
N/A

### Problem description
A strategy to help handle large weights for Gemma-2-27b is to [add the following shard specs](https://github.com/tenstorrent/tt-xla/issues/1494#issuecomment-3575898877):
```
        shard_specs[model.model.embed_tokens.weight] = ("batch", "model")
        shard_specs[model.lm_head.weight] = ("batch", "model")
```
[Edit]: or `(None, "batch")` for mesh_shape `(2, 4)`

 We want to add unit tests for sharding these ops (`nn.Embedding`,`nn.Linear`) for this model. The op tester infra doesn't support passing in `mesh` or `shard_spec_fn` yet.
 
### What's changed
- Added support for `mesh` and `shard_spec_fn` as optional inputs to `run_op_test()` (same as done in `run_graph_test()`) and updated `test_spmd_sharding` to use the op_tester.

- Added tensor-parallel op tests for Gemma-2-27b `lm_head` and `embed_tokens` [Edit]: and parametrized by `mesh_shape` and `shard spec` to test both possible shard specs.

### Checklist
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by kmabeeTT** [comment]
Nice tests and changes! Those shard specs you linked to from my comment were an initial guess (then there has been a lot of offline discussion about them and debug with @hshahTT, trying alternatives but no hard conclusions), I think it makes sense to go with this for now though, unless Het has alternative that is worth using.

**Comment by hshahTT** [comment]
Oh that happens because it's a (1, 8) mesh with "batch" = 1 and "model" = 8. So `(None, "batch")` is the same as `(None, None)` (since the batch axis only has 1 device).

Could you change this to a (2, 4) mesh with "batch" = 2 and "model" = 4 instead?

**Comment by gengelageTT** [comment]
Tests passing with new parametrization by mesh_shape here: https://github.com/tenstorrent/tt-xla/actions/runs/20142144630

**Comment by kmabeeTT** [comment]
Unfortunately mesh_shape paramaterization on test (mixing mesh shapes in same CI job) will not work after imminent transformers uplift [PR](https://github.com/tenstorrent/tt-xla/pull/2545) according to Jona, which brings an assert on mixing not being allowed, and explicit change to separate models into separate CI jobs to not mix 1x8 and 2x4 mesh_shape models.   

To avoid breakage once his change lands, we'd have to do the same here (split tests into separate CI jobs), which is annoying... or temporarily stick to the same mesh_shape across the board for all tests in same job while this is an issue (pending resolution of https://github.com/tenstorrent/tt-xla/issues/2308, not been looked into too deep yet).
